src/Optimizely/DecisionService/DecisionService.php

```diff
@@ -87,23 +87,19 @@ public function __construct(LoggerInterface $logger, ProjectConfig $projectConfi
      * @param string $userId         user ID
      * @param array  $userAttributes user attributes
      *
-     * @return string  the bucketing ID assigned to user
+     * @return String representing bucketing ID if it is a String type in attributes else return user ID.
      */
     protected function getBucketingId($userId, $userAttributes)
     {
-        // By default, the bucketing ID should be the user ID
-        $bucketingId = $userId;
-
-        // If the bucketing ID key is defined in userAttributes, then use that in
-        // place of the userID for the murmur hash key
         $bucketingIdKey = ControlAttributes::BUCKETING_ID;
-        
-        if (!empty($userAttributes[$bucketingIdKey]) && is_string($userAttributes[$bucketingIdKey])) {
-            $bucketingId = $userAttributes[$bucketingIdKey];
-            $this->_logger->log(Logger::DEBUG, sprintf('Setting the bucketing ID to "%s".', $bucketingId));
-        }
 
-        return $bucketingId;
+        if (isset($userAttributes[$bucketingIdKey]) && !is_null($userAttributes[$bucketingIdKey])){
+            if (is_string($userAttributes[$bucketingIdKey])){
```

Comment:
I think you don't need the 2nd part: `**!is_null($userAttributes[$bucketingIdKey]**`
isset will also check that the value is not set to NULL. Please confirm. 
If this is the case, please merge isset and is_string in one if.